packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriodForm.component.js

```diff
@@ -0,0 +1,164 @@
+import { useEffect, useState } from 'react';
+
+import { subDays } from 'date-fns';
+import PropTypes from 'prop-types';
+
+import {
+	ButtonPrimary,
+	ButtonTertiary,
+	Form,
+	InlineMessageDestructive,
+	SizedIcon,
+	StackHorizontal,
+	StackVertical,
+} from '@talend/design-system';
+import { InputDateTimeRangePicker } from '@talend/react-components/lib/DateTimePickers';
+
+import { getDataAttributesFrom } from '../../../helpers/usage.helpers';
+
+function getPeriodOptions(t) {
+	return [
+		{
+			id: 'LAST_24_HOURS',
+			value: 'lastDay',
+			label: t('LAST_24_HOURS', { defaultValue: 'Last 24 hours' }),
+		},
+		{
+			id: 'LAST_3_DAYS',
+			value: 'last3days',
+			label: t('LAST_3_DAYS', { defaultValue: 'Last 3 days' }),
+		},
+		{
+			id: 'LAST_WEEK',
+			value: 'last7days',
+			label: t('LAST_WEEK', { defaultValue: 'Last 7 days' }),
+		},
+		{
+			id: 'LAST_MONTH',
+			value: 'last30days',
+			label: t('LAST_MONTH', { defaultValue: 'Last 30 days' }),
+		},
+		{
+			id: 'CUSTOM',
+			value: 'custom',
+			label: t('CUSTOM', { defaultValue: 'Custom' }),
+			checked: false,
+		},
+	];
+}
+
+const BadgePeriodForm = ({ id, onChange, onSubmit, t, value, ...rest }) => {
+	const [isCustom, setCustom] = useState(value.id === 'CUSTOM');
+	const badgePeriodFormId = `${id}-period-form`;
+	const goBack = () => setCustom(false);
+	const [dirty, setDirty] = useState(false);
+	const initialStartDateTime = subDays(new Date(), 1);
+	const initialEndDateTime = new Date();
+
+	const resetRange = () => {
+		onChange(null, {
+			startDateTime: initialStartDateTime,
+			endDateTime: initialEndDateTime,
+			errorMessage: null,
+		});
+	};
+
+	useEffect(() => {
+		if (dirty) {
+			onSubmit();
+			setDirty(false);
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [dirty]);
+	return (
+		<Form id={`${badgePeriodFormId}-form`} onSubmit={onSubmit}>
+			{!isCustom && (
+				<StackVertical gap="0" align="normal">
+					{getPeriodOptions(t).map(rowItem => {
+						return (
+							<ButtonTertiary
+								key={rowItem.id}
+								onClick={event => {
```

Comment:
Maybe use hook useCallback could be good optimization for this event listener building